src/components/Form/Form.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { useForm, FormProvider } from "react-hook-form";
+import submitButton from "../../images/submitbutton.svg";
+
+export type FormValues = {
+  description: string;
+  agreeToTerms: boolean;
+};
+
+interface FormProps {
+  submit: (data: FormValues) => void;
+  children?: React.ReactNode;
+  className?: string; // Add className for dynamic styling
+}
+
+const Form: React.FC<FormProps> = ({ submit, children, className }) => {
+  const methods = useForm<FormValues>({
+    mode: "onBlur",
+  });
+
+  const onSubmit = methods.handleSubmit((data) => {
+    console.log("Form submitted with data:", data);
+    submit(data);
```

Comment:
When I submit the form in Storybook, I don't see anything logged under the Actions tab, is there a way we can make that work in this scenario?

Reply:
Just finished that right now !